src/azure-cli-core/azure/cli/core/azclierror.py

```diff
@@ -25,9 +26,11 @@ class AzCLIError(CLIError):
     """ Base class for all the AzureCLI defined error classes.
     DO NOT raise this error class in your codes. """
 
-    def __init__(self, error_msg, recommendation=None):
+    def __init__(self, error_msg, command=None, recommendation=None):
```

Comment:
Consider adding `command` parameter after `recommendation` to avoid unexpected error caused by instantiating `AzCLIError` with `AzCLIError("error message", "recommendation")`, as `"recommendation"` will now be parsed as `command`.

Even though it is a better practice to write `AzCLIError("error message", "recommendation")` as `AzCLIError("error message", recommendation="recommendation")`, we can't guarantee everyone does this.